synapse/media/media_repository.py

```diff
@@ -912,7 +912,11 @@ async def _federation_download_remote_file(
                 )
                 raise SynapseError(502, "Failed to fetch remote media")
 
-            if b"Content-Type" in headers:
+            if (
+                b"Content-Type" in headers
+                and len(headers[b"Content-Type"]) > 0
+                and headers[b"Content-Type"][0] is not None
```

Comment:
We could apply extra scrutiny via

```suggestion
                and isinstance(headers[b"Content-Type"][0], str)
```

Reply:
Good point! I believe `headers[b"Content-Type"][0]` is `bytes`, but the point of `isinstance` still stands.